homeassistant/components/integration/sensor.py

```diff
@@ -105,8 +105,8 @@ def __init__(
         self._name = name if name is not None else f"{source_entity} integral"
 
         if unit_of_measurement is None:
-            self._unit_template = "{}{}{}".format(
-                "" if unit_prefix is None else unit_prefix, "{}", unit_time
+            self._unit_template = (
+                f"{'' if unit_prefix is None else unit_prefix}{'{}'}{unit_time}"
```

Comment:
This of course works, but you leave another occurrence behind:

homeassistant/components/integration/sensor.py#L141

Reply:
Yeah, I was not sure how to handle line 141. Should I combine the two into one in line 108-109?

Reply:
in L108 we don't know the unit yet, so it must be postponed to L141

Don't know how to handle this with f-strings :(

Reply:
Unfortunately, me neither :(

Reply:
I think this is one of those places where following the rules to make the code better is actually going to make the code worse. At least unless a bit more refactoring is done first. I think we are better off trying to get rid of the double format in the first place.

How about:

 * Store the values in `self._unit_prefix` and `self._unit_time` without converting them first.
 * Get rid of the logic in `__init__` for `self._unit_of_measurement`  just store the value as is. Yes, get rid of `_unit_template` entirely.
 * In `async_added_to_hass` convert the unit prefix and unit time using `UNIT_PREFIXES` and `UNIT_TIME` as local variables. Don't set them on self at all and let the `calc_integration` function capture them from local state rather than from self.
 * The code for setting `self._unit_of_measurement` inside `async_added_to_hass` can now use f-strings as all the information is available there directly.

Reply:
why not update it but escape the characters.
```
f"{'' if unit_prefix is None else unit_prefix}{{}}{unit_time}"
```

Reply:
I would prefer to use the last option and escape the characters to keep the focus of this PR.

Reply:
Fair enough. I think the spirit of f-strings is to make code more legible, concise and easier to maintain so i would disagree, but not my call :-)

Reply:
Completely fair, I think I will do a PR at the end to clean up cases like this one :)